cloud-agent/service/server/src/main/scala/org/hyperledger/identus/issue/controller/IssueControllerImpl.scala

```diff
@@ -92,16 +96,31 @@ class IssueControllerImpl(
                 .mapError(_ =>
                   ErrorResponse.badRequest(detail = Some("Missing request parameter: credentialDefinitionId"))
                 )
-              credentialDefinitionId = {
+              credentialDefinition <- credentialDefinitionService.getByGUID(credentialDefinitionGUID)
+              credentialDefinitionId <- {
 
+                // TODO: this needs to be either DID or HTTP url based on credentialDefinition.resolutionMethod
                 val publicEndpointServiceName = appConfig.agent.httpEndpoint.serviceName
                 val resourcePath =
                   s"credential-definition-registry/definitions/${credentialDefinitionGUID.toString}/definition"
-                val didUrl = Url
-                  .parse(s"$issuingDID?resourceService=$publicEndpointServiceName&resourcePath=$resourcePath")
-                  .toString
 
-                didUrl
+                val didUrl = for {
+                  canonicalized <- JsonUtils.canonicalizeToJcs(credentialDefinition.definition.toJson)
+                  encoded = Base64Utils.encodeURL(canonicalized.getBytes)
+                  hash = Sha256Hash.compute(encoded.getBytes).hexEncoded
+                  didUrl = DIDUrl(
+                    issuingDID.did,
+                    Seq(),
+                    ListMap(
+                      "resourceService" -> Seq(publicEndpointServiceName),
+                      "resourcePath" -> Seq(s"credential-definition-registry/definitions/${credentialDefinitionGUID.toString}/definition?resourceHash=$hash"),
```

Comment:
Should the prefix `cloud-agent` be added to the path?
It should be a matter of configuration, so the agent should know the full external path